tests: net: tx_timestamp move to new ztest API

hunterhu3000 opened the pull request:
Move tx_timestamp tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

enjiamai commented on tests/net/tx_timestamp/src/main.c:
```diff
@@ -277,7 +277,7 @@ static void test_timestamp_setup_all(void)
 }
 
-static void test_timestamp_cleanup(void)
+static void test_timestamp_cleanup(void *dummy)
```
Why do we add this?

hunterhu3000 replied:
the tear down function needs a parameter of void *

nashif commented on tests/net/tx_timestamp/src/main.c:
```diff
@@ -486,17 +487,13 @@ static void test_check_timestamp_after_enabling(void)
 }
 
-void test_main(void)
+ZTEST(tx_timestamp_test_suite, test_tx_timestamp)
```
no need for _test_suite here.

hunterhu3000 replied:
Updated. Thanks to review.

hunterhu3000 committed:
tests: net: tx_timestamp move to new ztest API

Move tx_timestamp tests to use new ztest API
TEST=twister -T tests/net/tx_timestamp

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
The 	test_timestamp_setup_all and test_check_timestamp_after_enabling depends on the previous functions and need to be put in one function.
CONFIG_ZTEST_SHUFFLE=y can be set in this test.

hunterhu3000 commented:
@nashif Could you please help to review and +1, so that the PR can be merged in to the mainline? Thanks

carlescufi commented:
@hunterhu3000 please rebase

nashif commented:
this change is not needed anymore, it was already done in main branch.
